percona/controllers/version/version.go

```diff
@@ -16,29 +17,43 @@ func EnsureVersion(clientset kubeapi.Interface, cr *api.PerconaPGCluster, vs Ver
 	if cr.Spec.UpgradeOptions == nil {
 		return nil
 	}
+	var pVer string
 	pgCluster, err := clientset.CrunchydataV1().Pgclusters(cr.Namespace).Get(context.TODO(), cr.Name, metav1.GetOptions{})
-	if err != nil {
+	if err != nil && !kerrors.IsNotFound(err) {
 		return errors.Wrap(err, "get pgcluster resource")
 	}
-	pVer, err := GetPostgresqlVersin(clientset, pgCluster)
-	if err != nil {
-		return errors.Wrap(err, "get postgrsql version")
+	if pgCluster != nil && err == nil {
```

Comment:
rather than checking `err == nil` and doing the job inside the if, IMO it's better to return error if `err != nil` and unindent `GetPostgresqlVersion` call

Reply:
we need to continue here because it's ok if there no  pgClusster yet